src/main/java/org/spongepowered/common/registry/type/world/WeatherRegistryModule.java

```diff
@@ -24,39 +24,17 @@
  */
 package org.spongepowered.common.registry.type.world;
 
-import static com.google.common.base.Preconditions.checkNotNull;
-
-import com.google.common.collect.ImmutableList;
-import com.google.common.collect.Maps;
-import org.spongepowered.api.registry.CatalogRegistryModule;
-import org.spongepowered.api.registry.util.RegisterCatalog;
 import org.spongepowered.api.world.weather.Weather;
-import org.spongepowered.api.world.weather.Weathers;
+import org.spongepowered.common.registry.type.MutableCatalogRegistryModule;
 import org.spongepowered.common.weather.SpongeWeather;
 
-import java.util.Collection;
-import java.util.Map;
-import java.util.Optional;
-
-public final class WeatherRegistryModule implements CatalogRegistryModule<Weather> {
-
-    @RegisterCatalog(Weathers.class)
-    private final Map<String, Weather> weatherMappings = Maps.newHashMap();
-
-    @Override
-    public Optional<Weather> getById(String id) {
-        return Optional.ofNullable(this.weatherMappings.get(checkNotNull(id).toLowerCase()));
-    }
-
-    @Override
-    public Collection<Weather> getAll() {
-        return ImmutableList.copyOf(this.weatherMappings.values());
-    }
+public final class WeatherRegistryModule extends MutableCatalogRegistryModule<Weather> {
```

Comment:
SpongeWeather automatically prefixes the id with minecraft: making it inaccessible by the gr.

See http://pastebin.com/uNTjiecE line 356